log4j-appender/src/test/java/org/apache/kafka/log4jappender/KafkaLog4jAppenderTest.java

```diff
@@ -130,55 +116,51 @@ public void testLog4jAppends() {
         for (int i = 1; i <= 5; ++i) {
             logger.error(getMessage(i));
         }
-
-        Assert.assertEquals(
-            5, (getMockKafkaLog4jAppender()).getHistory().size());
+        assertEquals(getMockKafkaLog4jAppender().getHistory().size(), 5);
     }
 
-    @Test(expected = RuntimeException.class)
-    public void testLog4jAppendsWithSyncSendAndSimulateProducerFailShouldThrowException() {
+    @Test
+    public void testSyncSendAndSimulateProducerFailShouldThrowException() {
         Properties props = getLog4jConfig(true);
         props.put("log4j.appender.KAFKA.IgnoreExceptions", "false");
         PropertyConfigurator.configure(props);
 
         MockKafkaLog4jAppender mockKafkaLog4jAppender = getMockKafkaLog4jAppender();
         replaceProducerWithMocked(mockKafkaLog4jAppender, false);
 
-        logger.error(getMessage(0));
+        assertThrows(RuntimeException.class, () -> logger.error(getMessage(0)));
     }
 
     @Test
-    public void testLog4jAppendsWithSyncSendWithoutIgnoringExceptionsShouldNotThrowException() {
+    public void testSyncSendWithoutIgnoringExceptionsShouldNotThrowException() {
         Properties props = getLog4jConfig(true);
         props.put("log4j.appender.KAFKA.IgnoreExceptions", "false");
         PropertyConfigurator.configure(props);
 
         MockKafkaLog4jAppender mockKafkaLog4jAppender = getMockKafkaLog4jAppender();
         replaceProducerWithMocked(mockKafkaLog4jAppender, true);
 
-        logger.error(getMessage(0));
+        assertDoesNotThrow(() -> logger.error(getMessage(0)));
```

Comment:
What is the reason for having `assertDoesNotThrow` here and below? The test will fail if an exception is thrown, so seems like unnecessary noise.

Reply:
I considered it is a kind of highlight. However, it is unnecessary in this test case as it is the final line. @g1geordie WDYT? Should we file a PR to remove this unnecessary noise?

Reply:
As you think .  I just point out where we do the check .
I will file a MINOR to change that    . Thank you